### DISCUSSION TITLE
How can I hide a constant's value?

### DISCUSSION THREAD

**Discussion by mblumtritt** [lsegal/yard#1580] [General]
Sometimes I define a constant which should be described but I don't like to document the value. I found two cases: a) the constant value is somehow computed or b) too big.

**Question:** Is there any way to "hide" the value but keep documentation text for such constants?

#### sample a) computed constant; taken from [Rake source](https://github.com/ruby/rake/blob/c6829ffcdfab370d1b5fe788f9ce928f81a8ac56/lib/rake/file_utils.rb#L10-L13)
```ruby
RUBY = ENV["RUBY"] || File.join(
  RbConfig::CONFIG["bindir"],
  RbConfig::CONFIG["ruby_install_name"] + RbConfig::CONFIG["EXEEXT"]).
  sub(/.*\s.*/m, '"\&"')
```

#### sample b) big constant
```ruby
HELP = <<~TEXT
  ... here comes some multi-line text
TEXT
```

**Comment by lsegal** [comment]
You can use the `@!parse` directive to override the value, but you would need to show something as YARD currently does not support not having const values. It would be useful to display something declarative like "HIDDEN" or "INTERNAL" to indicate that the const value is explicitly hidden:

```ruby
# Documentation for const...
# @!parse
#   CONST = INTERNAL
CONST = 1
```

**Reply by mblumtritt** [reply]
Nice.
```ruby
# Documentation for const...
# @!parse
#   CONST = …
CONST = 'THIS WORKS'
```
